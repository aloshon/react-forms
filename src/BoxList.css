.boxlist{
    position: relative;
    margin: 0 auto;
    width: 800px;
    height: 800px;
    border: solid 1px black;
}