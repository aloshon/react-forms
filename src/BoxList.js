import React, {useState} from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import {v4 as uuid} from "uuid";
import './BoxList.css';

const BoxList = () => {
    const intitialState = []
    const [boxes, setBoxes] = useState(intitialState);

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }

    const remove = (box) => {
        setBoxes(boxes.filter(b => b.id !== box))
    }

    return (
        <div>
            <h3>Box List</h3>
            <NewBoxForm addBox={addBox}/>
            <div className="boxlist">
                {boxes.map(({id, color, width, height}) => <Box key={id} id={id} color={color} width={width} height={height} remove={remove}/>)}
            </div>
        </div>
    )
}

export default BoxList;