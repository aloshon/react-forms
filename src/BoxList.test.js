import React from "react";
import {render, fireEvent} from "@testing-library/react";
import BoxList from "./BoxList";

it("renders successfully", function (){
    render(<BoxList/>);
});

it("matches snapshot", function (){
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new Box", function() {
    // Checks to see if remove button is on DOM, if so then the box is rendered

    const { getByLabelText, queryByText, container } = render(<BoxList />);
    const colorInput = getByLabelText("Color");
    const heightInput= getByLabelText("Height");
    const widthInput = getByLabelText("Width");
    fireEvent.change(colorInput, { target: { value: "blue" } });
    fireEvent.change(widthInput, { target: { value: 100 } });
    fireEvent.change(heightInput, { target: { value: 100 } });
    const button = queryByText("Add Box!");
    fireEvent.click(button);

    const removeBtn = queryByText("X");
    expect(removeBtn).toBeInTheDocument();
  });