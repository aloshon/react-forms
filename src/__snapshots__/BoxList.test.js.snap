// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Box List
    </h3>
    <form>
      <label
        for="boxcolor"
      >
        Color
      </label>
      <input
        id="boxcolor"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
      <p />
      <label
        for="boxheight"
      >
        Height
      </label>
      <input
        id="boxheight"
        max="500"
        min="0"
        name="height"
        placeholder="px"
        type="number"
        value=""
      />
      <label
        for="boxwidth"
      >
        Width
      </label>
      <input
        id="boxwidth"
        max="500"
        min="0"
        name="width"
        placeholder="px"
        type="number"
        value=""
      />
      <button>
        Add Box!
      </button>
    </form>
    <div
      class="boxlist"
    />
  </div>
</DocumentFragment>
`;
